lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -641,12 +622,13 @@ bool mlir::triton::preProcessLoopAndGetSchedule(
   options.predicateFn = predicateOp;
   options.supportDynamicLoops = true;
   unsigned numLoadsInStage = (numStages - 2) * loads.size();
-  options.annotateFn =
-      [numLoadsInStage](Operation *op,
-                        mlir::triton::PipeliningOption::PipelinerPart part,
-                        unsigned iteration) {
-        return setWaitNum(op, part, iteration, numLoadsInStage);
-      };
+  options.annotateFn = [](Operation *op,
+                          mlir::triton::PipeliningOption::PipelinerPart part,
+                          unsigned iteration) {
+    if (isa<ttg::ExtractSliceOp>(op)) {
+      op->setAttr("triton.pipeline.need_wait", UnitAttr::get(op->getContext()));
```

Comment:
can you put the string in a constexpr so that we don't have bugs because of string typos :)